### ISSUE TITLE
Debugging - Please, for the love of God

### ISSUE THREAD

**Issue by octav-s** [mage-ai/mage-ai#5201]
**Is your feature request related to a problem? Please describe.**
I have a lot of pipelines to maintain and create, that connect to MySQL, PostgreSQL, MongoDB and do various tasks.
I have been creating and fixing them using print-outs, but that is difficult and very time consuming.
I've tried to run code from command line or IDE, but it seems that it is linked to web interface or my skills/time are not sufficient for the task.
Please add debug functionality. Something like Jupyter notebooks would be amazing, but anything will do.

**Describe the solution you'd like**
Debug functionality. Main things to consider:
- some kind of breakpoint functionality (execute next statement + continue until next breakpoint buttons etc)
- console available at break point in order to inspect

**Describe alternatives you've considered**
I'm currently printing out from mage source files from docker when I need to debug.

**Comment by PrashanthEns** [comment]
Yes.Debug would be great Feature.
Without debug very time taking to code and check issues.
I have raised this is slack as well
https://mageai.slack.com/archives/C03HTTWFEKE/p1718382846669249
![image](https://github.com/mage-ai/mage-ai/assets/97177463/cab4bcfe-e851-4b56-b439-2d8b62da6c44)

**Comment by shouldsee** [comment]
something like web_pdb? https://github.com/romanvm/python-web-pdb

**Comment by tommydangerous** [comment]
this is a great idea, looking into it

**Comment by tommydangerous** [comment]
yes, I agree with you. I’ve ran into so many inefficiencies debugging. There are some new features coming that’ll make life much more enjoyable.

**Comment by jamesgade-ensar** [comment]
@tommydangerous 
Our team has been facing huge blockers due to unavailability of debugging, I'm glad you've taken up this as priority.
Please keep us updated on this feature.

**Comment by octav-s** [comment]
Hi @tommydangerous, any updates on debugging? Can you at least let us know if there is a chance of getting it anytime soon? Thanks!
